.github/workflows/build.yaml

```diff
@@ -0,0 +1,59 @@
+name: build_and_test
+
+on:
+  push:
+    branches:
+      - main
+    tags:
+      - "*"
+  pull_request:
+
+jobs:
+  build_and_test:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+        with:
+          # Need to clone everything for the git tags.
+          fetch-depth: 0
+
+      - name: Set up Python
+        uses: actions/setup-python@v4
+        with:
+          python-version: "3.10"
+          cache: "pip"
+
+      - name: Install yaml
+        run: sudo apt-get install libyaml-dev
+
+      - name: Install prereqs for setuptools
+        run: pip install wheel
+
+      - name: install peigen to access EIGEN headers
+        run: pip install peigen
+
+      - name: Install dependencies
+        run: pip install -r requirements.txt
+
+      # We have two cores so we can speed up the testing with xdist
+      - name: Install xdist, openfiles and flake8 for pytest
+        run: >
+          pip install pytest-xdist pytest-openfiles pytest-flake8
+          pytest-cov "flake8<5"
+
+      - name: Build and install
+        run: pip install -v -e .
+
+      - name: Install documenteer
+        run: pip install 'documenteer[pipelines]<0.7'
```

Comment:
No need to install documenteer in this action

Reply:
It was failing before it added that. I think because I test the docs as part of my pytest to keep the code sections from bit rotting.

Reply:
Ah. Okay. I saw you were building docs in a different action.

Reply:
Yeah, but AFAIK documenteer doesn't test that the code blocks will run, right? If it does I can remove those options here.